src/services/rpc.rs

```diff
@@ -19,20 +21,24 @@ use crate::services::{
     pb::{psh_service_client::PshServiceClient, HostInfoRequest},
 };
 
+use super::config::RpcConfig;
+
 #[derive(Clone, Debug)]
 pub struct RpcClient {
     token: String,
     client: PshServiceClient<tonic::transport::Channel>,
     raw_info: RawInfo,
+    duration: Duration,
 }
 
 impl RpcClient {
-    pub async fn new(addr: &str, token: String) -> Result<Self> {
+    pub async fn new(config: RpcConfig) -> Result<Self> {
         let client: PshServiceClient<tonic::transport::Channel> =
-            PshServiceClient::connect(format!("https://{}", addr)).await?;
```

Comment:
This change should be in a separate commit since it's not relevant to your commit message.